### ISSUE TITLE
Write OEP answering "What code is part of the Open edX project?"

### ISSUE THREAD

**Issue by sarina** [openedx/open-edx-proposals#295]
* What organizations do (and can and should) hold code that belongs to the project?
* How do we govern code moving in and out of the project? (note: OEP-14 Archiving Repos touches on the mechanics of repo transfer out of the `openedx` org).
  * Are community organization repos governed by OEPs?
  * How to handle repos moving into the Open edX project that have not followed OEP guidelines?

**Comment by kdmccormick** [comment]
Thanks for capturing this 🔥

**Comment by robrap** [comment]
I've been working in [eduNEXT/openedx-events](https://github.com/eduNEXT/openedx-events), and it serves as a good example.

I believe `eduNEXT/openedx-events` is meant to be a core part of the Open edX platform, but it isn't yet in the `openedx` repo.
1. Does that mean it isn't officially core to the platform until it moves orgs?
2. Does it mean that OEPs don't need to be enforced until it moves?
3. How would I request that it be moved?
4. What would happen if eduNEXT didn't want it to move?

I imagine this will be a recurring issue, as it will be quicker for any company to start a repo under its own org, even though it may be better for it to live in the `openedx` org. This is especially true for edX, who has created the most repos to date.

**Comment by kdmccormick** [comment]
One indicator for `openedx` org inclusion that I think @ormsbee had floated was "If the default Open edX installation cannot run without it, it needs to be in the `openedx` org." I think this is a good idea. Notes:
* The converse doesn't necessarily apply: the `openedx` org will contain some non-essential components.
* Of course, this does not third-party libraries like Django.

> Does that mean it isn't officially core to the platform until it moves orgs?

To me, yes. "Core" code should exist in one place.

> Does it mean that OEPs don't need to be enforced until it moves?

Again, yes. But it shouldn't be moved (and, therefore, shouldn't be required by the default installation) until follows OEPs, at least to a degree deemed reasonable.

> How would I request that it be moved?

Not sure. I'd say "make a request to tCRIL", but I'm not sure whether or not tCRIL should be the sole deciders of what goes in the org. Interested to hear this discussion on the OEP.

> What would happen if $FIRM didn't want it to move?

Well, the code should be under a free license, so we can fork any code into organization if we really must.

That being said, when working with the community, we should make this OEP's policies clear early on and make sure we're on the same page about the end state of the repo. Ideally, we should avoid anyone on either side of this process being surprised.

Also, I forget if we codified this somewhere or not, but bringing a repo into the openedx org will almost always result in the principal developer(s) of that code becoming core contributors with maintainer access to the moved repo. Following that guideline should reduce reluctance about the openedx org taking in new code.

> I imagine this will be a recurring issue, as it will be quicker for any company to start a repo under its own org, even though it may be better for it to live in the openedx org. This is especially true for edX, who has created the most repos to date.

That is OK with me. I think it is natural for Open edX components to start their lifecycle within a firm's organization, only being installed into the platform as optional dependencies, until we deem the code "ready" to move into the core `openedx` org.

**Comment by sarina** [comment]
> Also, I forget if we codified this somewhere or not, but bringing a repo into the openedx org will almost always result in the principal developer(s) of that code becoming core contributors with maintainer access to the moved repo. Following that guideline should reduce reluctance about the openedx org taking in new code.

I don't think it belongs in the CC OEP-54, because I think of that as almost a "how to be a CC" manual; the route of "making a bunch of code and having it moved to the org" falls under this OEP with a mention worthy in the maintainers OEP-55.

**Comment by robrap** [comment]
> "If the default Open edX installation cannot run without it, it needs to be in the openedx org."

@kdmccormick: I think this comment and your notes are a great start. You'll still need to consider exceptions like any repos that have not been moved to `openedx`, but should have been based on your notes. Or, any new dependencies that slip through the cracks and are not optional. Also, you'll need to cover how to deal with MFEs, or new top-level services. Would it simply cover anything tagged to be included in the named release? Other?

**Comment by kdmccormick** [comment]
Started on a very early draft: https://github.com/openedx/open-edx-proposals/pull/312

**Comment by kdmccormick** [comment]
Putting the draft down for now to focus on the Tutor work.

**Comment by kdmccormick** [comment]
Overdue update: This will involve two OEPs (57 and 61), as well as several edits to existing OEPs. I wrote up an overview of the plan here: https://openedx.atlassian.net/wiki/spaces/COMM/pages/3511091214/OEP-57+OEP-61+Update+Game+Plan

OEP-57 has entered review already: https://openedx.atlassian.net/wiki/spaces/COMM/pages/3540713547/Open+edX+Proposal+57+Product+Offering

Once OEP-57 is merged, I/we can start on OEP-61 and the other OEP updates.

**Comment by sarina** [comment]
Update: OEP-57 is merged, but as discussed in https://github.com/openedx/open-edx-proposals/pull/393, we're in a holding pattern until we have firmer definition of Core Product. @kdmccormick What would you think if I revived a bit of what was done in #312 as like, the first part of OEP-61 that covers some of this stuff that Axim has basically already codified (core repos (for now by the definition provided above:

> "If the default Open edX installation cannot run without it, it needs to be in the openedx org." 

) as well as the mechanism of transfer action (including perhaps an addition of a forums notification and/or RFC as well as IP process stuff the authors will need to follow and addition of core contributor status).

I think that would be a good, scoped start to cover concerns of this issue, while leaving space to fill in more as Core Product is defined.

**Comment by kdmccormick** [comment]
@sarina That sounds really reasonable to me!

**Comment by kdmccormick** [comment]
It's become clearer to me over time that "the Open edX project" == "the openedx GitHub organization", and thus OEPs apply to code within the openedx GitHub organization. Is there any disagreement? Do we need to encode that in its own OEP, or can we just add it as a note into OEP-1?

**Comment by robrap** [comment]
> It's become clearer to me over time that "the Open edX project" == "the openedx GitHub organization", and thus OEPs apply to code within the openedx GitHub organization. Is there any disagreement? Do we need to encode that in its own OEP, or can we just add it as a note into OEP-1?

1. Sounds generally reasonable to me.
2. I'd just update OEP-1 as appropriate. Sounds like a clarification.
3. This may be obvious, but your note only applies to code-related OEPs. There are also other types of OEPs.

**Comment by sarina** [comment]
I agree with Robert on all points. Thanks for updating this issue.

**Comment by kdmccormick** [comment]
Great. This pretty low in my backlog of work, so if any contributor is interested in shepherding this update through before I get to it, feel free!
